_docs-2/troubleshooting/code.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Codebase Overview
+category: troubleshooting
+order: 7
+---
+
+If you are trying to troubleshoot an Accumulo problem, you should first try and find help within Accumulo's documentation. The Accumulo
+website's [search tool](https://accumulo.apache.org/search/) can help in finding relevant documentation.
+
+If you cannot find relevant documentation for your problem, you might want to review Accumulo's code to learn more about the code that
+is causing the problem and find a potential solution for it. This documentation provides an overview of Accumulo's codebase to get you
+started.
+
+## Master
+
+The [Master] has the following responsibilities:
+
+  * [detect and respond][update] to [TabletServer] failures
+  * assign and balance tablets to Tablet Servers using a [TabletBalancer]
+  * handle table creation, alteration and deletion requests from clients using the [TableManager]
+  * coordinate changes to write-ahead logs using the [WalStateManager].
```

Comment:
Could be more specific.  It coordinates sorting of WALs and associates WALs to tablets that need them (tablets that were previously assigned to a dead tserver) before assigning.